docker/Dockerfile

```
@@ -56,7 +56,8 @@ COPY docker/docker-ros/docker/recursive_vcs_import.py /usr/local/bin
 RUN apt-get update && \
     apt-get install -y python-is-python3 && \
     rm -rf /var/lib/apt/lists/*
-RUN /usr/local/bin/recursive_vcs_import.py src src/upstream
+ARG ENABLE_RECURSIVE_CLONED="True"
+RUN /usr/local/bin/recursive_vcs_import.py src src/upstream ${ENABLE_RECURSIVE_CLONED}
```

I suggest to fall back to just calling `vcs import` directly, if recursion is disabled. Somewhat goes against the naming of `recursive_vcs_import.py` otherwise.

```suggestion
ARG ENABLE_RECURSIVE_CLONED="true"
RUN if [[ $ENABLE_RECURSIVE_ADDITIONAL_DEBS == 'true' ]]; then \
        /usr/local/bin/recursive_vcs_import.py src src/upstream ; \
    else \
        # vcs import ... ; \
    fi
```

---

Is it possible to include every *.repo files in the current directory just using the vcs import command? I could see the benefit of a repository having for instance "hardware.repo", "vision.repo", etc. Thus, when I added the flag I was still interested in the recursive search, just not for the cloned packages.

---

ping @lreiher

---

I have discussed this with my peers and we don't really see the necessity for multiple top-level `.repo` files that are all found. Either there is one that you would like to include, or you want to search recursively. The only additional argument that we could see being integrated is to have a variable for configuring the script to search for another filename, e.g., `vision.repo` instead of `.repos`. Similar to how you can configure `CUSTOM_SCRIPT_FILE`.